pkg/apis/core/validation/shoot.go

```diff
@@ -265,6 +265,13 @@ func ValidateShootSpec(meta metav1.ObjectMeta, spec *core.ShootSpec, fldPath *fi
 		} else if len(ptr.Deref(spec.SecretBindingName, "")) != 0 && len(ptr.Deref(spec.CredentialsBindingName, "")) != 0 {
 			allErrs = append(allErrs, field.Forbidden(fldPath.Child("secretBindingName"), "is incompatible with credentialsBindingName"))
 		}
+
+		// TODO(dimityrmirchev) Remove once we drop support for Kubernetes 1.34
+		// Forbid secretBindingName for Kubernetes versions >= 1.34
+		if spec.SecretBindingName != nil && len(*spec.SecretBindingName) > 0 &&
+			k8sVersion != nil && versionutils.ConstraintK8sGreaterEqual134.Check(k8sVersion) {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("secretBindingName"), "is no longer supported for Kubernetes >= 1.34, use spec.credentialsBindingName instead"))
+		}
```

Comment:
Do you already want to add logic for the forceful shoot update through the maintenance controller?

Reply:
WDYT about me adding this logic in a different PR? Since 1.34 is not yet even released I think we have time